### PR TITLE
Updated to use the new Yesterday's Pollen API

### PR DESCRIPTION
**Pull request by goibon**
Several changes have been introduced to accomodate the changes introduced in [#4](https://github.com/Tomorrows-pollen-today/yesterdays-pollen-today/pull/4) of [Yesterday's Pollen](https://github.com/Tomorrows-pollen-today/yesterdays-pollen-today/).

* `routes/api.js` has been updated to take pollentype and location as query parameters
* `routes/api.js` has been updated to call the `/api/pollen/{date}?pollentype={pollentype}&location={location}` endpoint of Yesterday's Pollen
* Added `middleware/pollenTypes.js`
* `public/javascript/index.js` has been updated to find all elements with a `data-pollen-type` attribute and fetch pollen numbers for that type
* Updated `views/index.pug` to render chart elements for each type of pollen
* Updated `views/index.pug` and `public/stylesheets/style.scss` to be capable of displaying multiple pollen types.

### CODE REVIEW

File: public/javascript/index.js [hunk 1]
```diff
@@ -1,26 +1,45 @@
...
+
+/**
+ * Fetches the predicted pollen count for tomorrow for the given pollen type and location.
+ * @param  {number} pollenType The type of pollen to get a prediction for.
+ * @param  {number} location   The location to get a prediction for.
+ * @return {Promise}           A promise object that is resolved with an object containing the predicted
+ *                             pollen count as well as the location.
+ */
+async function fetchPollen (pollenType, location) {
+  let requestUrl = new URL('http://localhost:3000/api/pollen')
```
Comment (lassevalentini) [inline public/javascript/index.js]: I think localhost is a mistake here? :)

**Review by lassevalentini — CHANGES_REQUESTED** [review]
Code looks great except for one minor localhost call :+1:
How mobile-friendly is the layout? My hunch is that we can fit two pollen types in now, but we will have to go responsive later. Could possibly be done with a sexy grid layout.

File: public/javascript/index.js (hunk 1 above)
Comment (goibon) [reply public/javascript/index.js]: Haha yes that was definitely a mistake, good catch 😃

**Comment by goibon** [comment]
> How mobile-friendly is the layout? My hunch is that we can fit two pollen types in now, but we will have to go responsive later. Could possibly be done with a sexy grid layout.

Well I have thought of that actually and I think we can just show the bars next to each other and have them scale with the amount of pollen types available. If you look at [index.pug:8](https://github.com/Tomorrows-pollen-today/tomorrows-pollen-today/pull/13/files#diff-e3e7b061156de43ef48216789661ef0eR8) you'll notice that I'm scaling each container with the amount of pollen types.
Here is an example of how it looks with more than the two, note however that I've altered the colors in the example, in our code all the bars are the same color. I'm thinking that coloring the bars based on pollen type will be a separate PR.
Perhaps we will need to rotate the pollen names, and maybe even the values if they get crazy (does birch sometimes have four digit numbers? 🤔 ) to make sure it fits.

TmrwsPollenResponsive

**Comment by lassevalentini** [comment]
You're right that there's plenty og space for more pollen types. 
Looks great! We've got at least a couple of years till we have to worry about real estate for bars.

**Review by lassevalentini — APPROVED** [review]

**Pull request merged**